src/js/ripple/keypairs.js

```diff
@@ -0,0 +1,380 @@
+  return rs;
+}
+
+function createAccountId(pubKeyBytes) {
+  var bits = sjcl.codec.bytes.toBits(pubKeyBytes);
+  var hash = sjcl.hash.ripemd160.hash(sjcl.hash.sha256.hash(bits));
+  var id = UInt160.from_bits(hash);
+  id.set_version(Base.VER_ACCOUNT_ID);
+  return id;
+}
+
+/* --------------------------------- KEYPAIR -------------------------------- */
+
+function KeyPair() {}
+
+/*
+@param {Array} message
+@virtual
+ */
+KeyPair.prototype.sign = isVirtual;
+
+
+/*
+@param {Array<Number>} message
+@param {Array<Number>} signature
+@virtual
+ */
+KeyPair.prototype.verify = isVirtual;
+
+/*
+@return {Array<Number>} of bytes, in canonical form, for signing
+@virtual
+*/
+KeyPair.prototype.pubKeyBytes = isVirtual;
+
+hasCachedProperty(KeyPair, 'pubKeyHex', function() {
+  return bytesToHex(this.pubKeyBytes());
+});
+
+hasCachedProperty(KeyPair, 'account', function() {
+  return createAccountId(this.pubKeyBytes());
+});
+
+KeyPair.prototype.signHex = function(message) {
+  return bytesToHex(this.sign(message));
+};
+
+/* ---------------------------- SECP256K1 KEYPAIR --------------------------- */
+
+/*
+* @class
+* @private
+* @param {Object} - obj secretKey/publicKey Uint8Array members
+*                   (as returned by nacl.sign.keyPair)
+*/
+function Ed25519Pair(keyPair) {
+  KeyPair.apply(this, arguments);
+  this.secretKey = keyPair.secretKey;
+  this.publicKey = keyPair.publicKey;
+  this.type = KeyType.ed25519;
+}
+
+util.inherits(Ed25519Pair, KeyPair);
+
+/**
+* @param {Seed} seed - A 128 bit seed
+* @return {Ed25519Pair} key pair
+*/
+Ed25519Pair.fromSeed = function(seed) {
+  var seed256 = deriveEdKeyPairSeed(seed);
+  var keyPair = nacl.sign.keyPair.fromSeed(seed256);
+  return new Ed25519Pair(keyPair);
+};
+
+hasCachedProperty(Ed25519Pair, 'pubKeyBytes', function() {
+  return [0xED].concat(toGenericArray(this.publicKey));
+});
+
+Ed25519Pair.prototype.sign = function(message) {
+  var messageArray = new Uint8Array(message);
+  var signatureArray = nacl.sign.detached(messageArray, this.secretKey);
+  return toGenericArray(signatureArray);
+};
+
+Ed25519Pair.prototype.verify = function(message, signature) {
+  var messageArray = new Uint8Array(message);
+  var signatureArray = new Uint8Array(signature);
+  return nacl.sign.detached.verify(messageArray, signatureArray,
+                                   this.publicKey);
+};
+
+/* ---------------------------- SECP256K1 KEYPAIR --------------------------- */
+
+/*
+* @class
+* @private
+*/
+function Secp256k1Pair(secretKey) {
+  KeyPair.apply(this, arguments);
+  this.secretKey = secretKey;
+  this.type = KeyType.secp256k1;
+  this.curve = sjcl.ecc.curves.k256;
+}
+
+util.inherits(Secp256k1Pair, KeyPair);
+
+Secp256k1Pair.fromSeed = function(seed) {
+  return new Secp256k1Pair(derivek256Secret(seed));
+};
+
+hasCachedProperty(Secp256k1Pair, 'pubKey', function() {
+  var exponent = this.secretKey._exponent;
+  var EcdsaPublicKey = sjcl.ecc.ecdsa.publicKey;
```

Comment:
Just put something like this at the top of the file
`/* eslint new-cap: [2, {newIsCapExceptions: ["secretKey", "publicKey"]}] */`